gcp-observability/src/main/java/io/grpc/gcp/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -69,82 +46,41 @@ public class ConfigFilterHelper {
-      int currentMessageBytes
-          = currentFilter.messageBytes != null ? currentFilter.messageBytes : 0;
-      if (methodOrServicePattern.equals("*")) {
-        // parse config for global, e.g. "*"
-        if (perServiceFilters.containsKey(globalPattern)) {
-          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
-          continue;
-        }
-        FilterParams params = FilterParams.create(true,
-            currentHeaderBytes, currentMessageBytes);
-        perServiceFilters.put(globalPattern, params);
-      } else if (methodOrServicePattern.endsWith("/*")) {
-        // TODO(DNVindhya): check if service name is a valid string for a service name
-        // parse config for a service, e.g. "service/*"
-        String service = MethodDescriptor.extractFullServiceName(methodOrServicePattern);
-        if (perServiceFilters.containsKey(service)) {
-          logger.log(Level.WARNING, "Duplicate entry : {0)", methodOrServicePattern);
-          continue;
-        }
-        FilterParams params = FilterParams.create(true,
-            currentHeaderBytes, currentMessageBytes);
-        perServiceFilters.put(service, params);
-      } else {
-        // TODO(DNVVindhya): check if methodOrServicePattern is a valid full qualified method name
-        // parse pattern for a fully qualified method, e.g "service/method"
-        if (perMethodFilters.containsKey(methodOrServicePattern)) {
-          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
-          continue;
+  /**
+   * Checks if the corresponding service/method passed needs to be logged according to user provided
+   * observability configuration.
+   * Filters are evaluated in text order, first match is used.
+   *
+   * @param fullMethodName the fully qualified name of the method
+   * @return FilterParams object 1. specifies if the corresponding method needs to be logged
+   *     (log field will be set to true) 2. values of payload limits retrieved from configuration
+   */
+  public FilterParams logRpcMethod(String fullMethodName, Boolean client) {
+    FilterParams params = NO_FILTER_PARAMS;
+
+    int index = checkNotNull(fullMethodName, "fullMethodName").lastIndexOf('/');
+    String serviceName = fullMethodName.substring(0, index);
+    List<LogFilter> logFilters =
+        client ? config.getClientLogFilters() : config.getServerLogFilters();
+
+    for (LogFilter logFilter : logFilters) {
+      if (logFilter.matchAll
+          || logFilter.services.contains(serviceName)
+          || logFilter.methods.contains(fullMethodName)) {
+        if (logFilter.excludePattern) {
+          return params;
         }
-        FilterParams params = FilterParams.create(true,
-            currentHeaderBytes, currentMessageBytes);
-        perMethodFilters.put(methodOrServicePattern, params);
+        int currentHeaderBytes = logFilter.headerBytes != null ? logFilter.headerBytes : 0;
```

Comment:
As commented elsewhere if you had used primitive types in `logFilter` definition null checking wouldn't be necessary here (and line 79)